sdk/src/main/java/io/horizen/account/api/rpc/types/EthereumTransactionView.java

```diff
@@ -79,5 +79,38 @@ public EthereumTransactionView(EthereumTransaction tx, EthereumReceipt receipt,
         blockNumber = BigInteger.valueOf(receipt.blockNumber());
         transactionIndex = BigInteger.valueOf(receipt.transactionIndex());
     }
+
+    // constructor used for the pending transactions
+    public EthereumTransactionView(EthereumTransaction tx) {
+        blockHash = Hash.ZERO;
+        blockNumber = null;
+        transactionIndex = null;
+        type = BigInteger.valueOf(tx.version());
+        nonce = tx.getNonce();
+        from = tx.getFromAddress();
+        to = tx.getToAddress();
+        value = tx.getValue();
+        input = tx.getData();
+        gas = tx.getGasLimit();
+        gasPrice = tx.getGasPrice();
```

Comment:
do we need to account `baseFee` here:
`gasPrice = tx.getEffectiveGasPrice(baseFee);`
?

Reply:
let's check it. Geth may get the pending block base fee (it is known)

Reply:
Shouldn't the transactions look exactly like what was submitted?
- `gasPrice` only for legacy TX
- omit for EIP1559 and only set priority fee and fee cap?

Should check the behavior of GETH though.

Reply:
In the GETH implementation the transaction `gasFeeCap` value is used for the `gasPrice` field of EIP1559 transactions: https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1353
The `gasFeeCap` corresponds to our `EthereumTransaction.maxFeePerGas`, containing information on the maximum price per unit of gas that the user is willing to pay for his transaction.

Following the flow of the `txpool_content` method, starting from [here](https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L163), we will see that for every pending transaction an `RPCTransaction` is retrieved with the function [NewRPCPendingTransaction](https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1359)
`NewRPCPendingTransaction` internally calls the function `newRPCTransaction` that will finally assign, for the pending transactions, the `gasPrice` field with the `gasFeeCap` value.

Moreover, in the `NewRPCPendingTransaction` function the block number and the base fee are retrieved from the current (latest) blockchain block.
These values are passed to the `newRPCTransaction` but they are never used because the block hash in our case is only initialized (ex. [here](https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1348))
It seems that the gas price for pending transactions is expected to be calculated based on the last block but in reality is not.